src/Compilers/Core/VBCSCompiler/ServerDispatcher.cs

```diff
@@ -71,6 +71,8 @@ public static int Main(string[] args)
             // VBCSCompiler is installed in the same directory as csc.exe and vbc.exe which is also the 
             // location of the response files.
             var compilerExeDirectory = AppDomain.CurrentDomain.BaseDirectory;
+            // Pipename should be passed as the first and only argument to the server process
+            var pipeName = args[0];
```

Comment:
We should validate this arg. For example, what if it's missing or mal-formed or "-?" or "/help". It's not inconceivable that someone might try to launch VBCSCompiler.exe manually.